gef.py

```diff
@@ -10482,22 +10482,86 @@ def lookup_command(self, cmd):
 
         return None
 
+@register_command
+class AliasesCommand(GenericCommand):
+    """Base command to add, remove, or list aliases"""
 
-class GefAliases(gdb.Command):
-    """List all custom aliases."""
+    _cmdline_ = "aliases"
+    _syntax_  = "{:s} (add|rm|list)".format(_cmdline_)
 
     def __init__(self):
-        super().__init__("aliases", gdb.COMMAND_OBSCURE, gdb.COMPLETE_NONE)
+        super().__init__(prefix=True)
         return
 
-    def invoke(self, args, from_tty):
-        self.dont_repeat()
+    def do_invoke(self, argv):
+        self.usage()
+        return
+
+@register_command
+class AliasesAddCommand(GenericCommand):
+    """Command to add aliases"""
+
+    _cmdline_ = "aliases add"
+    _syntax_  = "{0} [ALIAS] [COMMAND]".format(_cmdline_)
+    _example_ = "{0} scope telescope".format(_cmdline_)
+
+    def __init__(self):
+        super().__init__(complete=gdb.COMPLETE_LOCATION)
+        return
+
+    def do_invoke(self, argv):
+        if (len(argv) < 2):
+            self.usage()
+            return
+        GefAlias(argv[0], " ".join(argv[1:]))
+        GefSaveCommand().invoke(None, False)
+        return
+
+@register_command
+class AliasesRmCommand(GenericCommand):
+    """Command to remove aliases"""
+
+    _cmdline_ = "aliases rm"
+    _syntax_ = "{0} [ALIAS]".format(_cmdline_)
+
+    def __init__(self):
+        super().__init__(complete=gdb.COMPLETE_LOCATION)
+        return
+
+    def do_invoke(self, argv):
+        if (len(argv) != 1):
+            self.rm_usage()
+            return
+        found = False
+        for idx, obj in enumerate(__aliases__):
```

Comment:
This whole `for` loop + `if not found`  could be replaced by something simpler using `list.remove`:
```python
try:
  __aliases__.remove(argv[0])
except ValueError:
  err("{0} not found in aliases.".format(argv[0]))
```

Reply:
For this one, the remove method wouldn't work since `__aliases__` is a list of objects (not strings). I used the for loop since we need to find the object which has the `_alias` attribute equal to `argv[0]`. I was able to use `__aliases__ = [alias for alias in __aliases__ if alias._alias != argv[0]]` to achieve the same effect as the original for loop in the beginning, but I'm not sure if it is as readable. Let me know what you think!

Reply:
As @Grazfather always says, "better to ask for forgiveness than permission": my snippet was overly simplified for you to understand what I meant, but I believe `remove` is more explicit about the action we're trying to do (i.e. removing the alias). So a better (working) code would be 

```python
try:
  alias_to_remove = next(filter(lambda x: x._alias == argv[0], __aliases__))
  __aliases__.remove(alias_to_remove)
except (ValueError, StopIteration) as e:
  err("{0} not found in aliases.".format(argv[0]))
```

Reply:
list.remove should work, assuming the test for equality works.

Does `__aliases__` have to be a list? It makes sense that it be a dictionary with keys representing the input command. This also makes it easy to make sure you don't add 2 aliases to the same input.